src/substrate-lib/substrate-context.tsx

```diff
@@ -0,0 +1,377 @@
+import { createInterBtcApi } from '@interlay/interbtc-api';
+import { ApiPromise } from '@polkadot/api';
+import { web3Accounts, web3Enable, web3FromAddress } from '@polkadot/extension-dapp';
+import { InjectedAccountWithMeta, InjectedExtension } from '@polkadot/extension-inject/types';
+import { KeyringPair } from '@polkadot/keyring/types';
+import { TypeRegistry } from '@polkadot/types/create';
+import jsonrpc from '@polkadot/types/interfaces/jsonrpc';
+import type { DefinitionRpcExt } from '@polkadot/types/types';
+import { keyring } from '@polkadot/ui-keyring';
+import { Keyring } from '@polkadot/ui-keyring/Keyring';
+import { isTestChain } from '@polkadot/util';
+import * as React from 'react';
+import { useLocalStorage } from 'react-use';
+
+import { APP_NAME } from '@/config/relay-chains';
+import { SELECTED_ACCOUNT_LOCAL_STORAGE_KEY } from '@/config/wallets';
+
+import * as constants from '../constants';
+
+const UNABLE_TO_INITIALISE_OPTIONS_MORE_THAN_ONCE = 'Unable to initialise options more than once';
+
+enum KeyringStatus {
+  Idle = 'IDLE',
+  Loading = 'LOADING',
+  Ready = 'READY',
+  Error = 'ERROR'
+}
+
+enum ApiStatus {
+  Idle = 'IDLE',
+  ConnectInit = 'CONNECT_INIT',
+  Connecting = 'CONNECTING',
+  Ready = 'READY',
+  Error = 'ERROR',
+  Disconnected = 'DISCONNECTED'
+}
+
+enum ActionType {
```

Comment:
You could break down this `substrate-context` into a `context` folder. These could be the files:
- types.ts
- reducer.ts
- provider.tsx
- index.ts

This structure makes it easier to review and to maintain later.